package/MDAnalysis/analysis/diffusionmap.py

```diff
@@ -0,0 +1,307 @@
+does not represent a sample from the equilibrium. This violates a basic
+assumption of the diffusion map and the results shouldn't be interpreted for this reason.
+This tutorial shows how to use the diffusionmap function.
+First load all modules and test data ::
+
+   >>> import MDAnalysis
+   >>> import numpy as np
+   >>> import MDAnalysis.analysis.diffusionmap as diffusionmap
+   >>> from MDAnalysis.tests.datafiles import PSF,DCD
+
+Given an universe or atom group, we can calculate the diffusion map from
+that trajectory using :class:`DiffusionMap`:: and get the corresponding eigenvalues
+and eigenvectors
+
+   >>> u = MDAnalysis.Universe(PSF,DCD)
+   >>> dmap = diffusionmap.DiffusionMap(u)
+   >>> dmap.run()
+   >>> eigenvalues = dmap.eigenvalues
+   >>> eigenvectors = dmap.eigenvectors
+
+Classes
+-------
+
+.. autoclass:: DiffusionMap
+   :members:
+
+References
+---------
+
+If you use this Dimension Reduction method in a publication, please
+reference:
+..[Lafon1] Coifman, Ronald R., Lafon, Stephane (2006) Diffusion maps.
+Appl. Comput. Harmon. Anal. 21, 5–30.
+..[Clementi1] Rohrdanz, M. A, Zheng, W, Maggioni, M, & Clementi, C. (2013)
+Determination of reaction coordinates via locally scaled
+diffusion map. Journal of Chemical Physics.
+
+.. If you choose the default metric, this module uses the fast QCP algorithm
+[Theobald2005]_ to calculate the root mean square distance (RMSD) between
+two coordinate sets (as implemented in :func:`MDAnalysis.lib.qcprot.CalcRMSDRotationalMatrix`).
+When using this module in published work please cite [Theobald2005]_.
+
+
+"""
+from six.moves import range
+import logging
+
+import numpy as np
+
+from MDAnalysis.analysis import rms
+
+from .base import AnalysisBase
+
+logger = logging.getLogger("MDAnalysis.analysis.diffusionmap")
+
+
+class DistMatrix(AnalysisBase):
+    """ Calculate the pairwise distance between each frame in a trajectory using
+        a given metric
+
+    Attributes
+    ----------
+    atoms : AtomGroup
+        Selected atoms in trajectory subject to dimension reduction
+    d_matrix : array
+        Array of all possible ij metric distances between frames in trajectory
+    """
+
+    def _prepare(self):
+        logger.info('numframes {0}'.format(self.nframes))
+        self.d_matrix = np.zeros((self.nframes, self.nframes))
+
+    def _single_frame(self):
+        traj_index = self._ts.frame
+        i_ref = self.atoms.positions - self.atoms.center_of_mass()
+        # diagonal entries need not be calculated due to metric(x,x) == 0 in
+        # theory, _ts not updated properly. Possible savings by setting a
+        # cutoff for significant decimal places to sparsify matrix
+        for j in range(self.stop-1, self._ts.frame-1, -self.step):
+            self._ts = self._u.trajectory[j]
+            j_ref = self.atoms.positions-self.atoms.center_of_mass()
+            ij_result = self._metric(i_ref, j_ref, weights=self._weights)
```

Comment:
it would be nice to pass other arguments to the distance function as well. You should also document the API of the metric so that users can use their own metrics